core/container.go

```diff
@@ -321,8 +321,12 @@ func (container *Container) From(ctx context.Context, gw bkgw.Client, addr strin
 
 	container.FS = def.ToPB()
 
-	// merge config.Env with imgSpec.Config.Env
-	imgSpec.Config.Env = append(container.Config.Env, imgSpec.Config.Env...)
+	// Merge environment and labels from previous container to the new one.
+	// Ff the environment variable or label also exists in the new configuration,
+	// it will replace the old one.
+	imgSpec.Config.Env = mergeEnv(imgSpec.Config.Env, container.Config.Env)
+	imgSpec.Config.Labels = mergeMap(imgSpec.Config.Labels, container.Config.Labels)
+
```

Comment:
For consistency, we also have [`ExposedPorts`](https://github.com/opencontainers/image-spec/blob/77d92ec93c784d88e7c8349207948aaf6d430785/specs-go/v1/config.go#L29), with corresponding:
- `WithExposedPort`
- `WithoutExposedPort`

Reply:
Added in:
- https://github.com/dagger/dagger/pull/5158